server/src/main/java/org/elasticsearch/action/admin/indices/refresh/TransportUnpromotableReplicaRefreshAction.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.refresh;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.index.engine.Engine;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+public class TransportUnpromotableReplicaRefreshAction extends HandledTransportAction<
+    UnpromotableReplicaRefreshRequest,
+    ActionResponse.Empty> {
+    public static final String NAME = RefreshAction.NAME + "[u]";
```

Comment:
Didn't know we're going to explicitly use "[u]" for unpromotable replicas. Do we just do it here, or we plan to make this distinction in the action names, also somewhere else? Considering this action is only for unpromotable replicas, why is this even necessary rather than just "[s]"?

Reply:
I did not discuss this with anyone yet indeed. But good you're starting this conversation :)

I just saw that we used `[r]` for the promotable replicas, so I thought it would be good to distinguish the unpromotable ones with `[u]`. I did not want to use `[s]` because that's already used for the main refresh action when the action is forwarded to a primary -- so I think then the system would get confused for which action to apply. 

Alternatively, I could use a totally different action name without a suffix, e.g., `indices:admin/refresh_unpromotable`. Feedback is welcome!

Reply:
Understood. It seems we have to have the distinction in the name, anyway. But for the sake of consistency, I was wondering, whether starting to use "[u]" would also require us to make sure we keep up with it elsewhere, or even adopt it in existing tests/code (e.g. `testAllowsNodeActions`)? Giving it a new name also doesn't seem incorrect to me, since it is a new action anyway.

Reply:
I'm good with `[u]` for this. There's no strict requirements in this area and it's not very well documented, but I think this fits the pattern ok:

- `[s]` is the action at the shard level, i.e. targeting all shard copies with a particular `ShardId`.
- `[p]` is the action on the primary of that shard.
- `[r]` is the action on a (promotable) replica of that shard.

We don't really need a convention yet but if we see enough other examples of this it would be worth making a generic "broadcast to all the unpromotable shards" action, and IMO a `[u]` suffix would totally make sense for that.